### PR TITLE
[17.0] [MIG] account_commission: Migration to 17.0

### PR DESCRIPTION
**Pull request by Tisho99**
Supersedes #499

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]

**Review by Jaimermaccione — COMMENTED** [review]
Review on Runboat. Encountering this error while trying to access invoices:

Traceback (most recent call last):
  File "/opt/odoo/odoo/tools/cache.py", line 99, in lookup
    r = d[key]
  File "<decorator-gen-5>", line 2, in __getitem__
  File "/opt/odoo/odoo/tools/func.py", line 87, in locked
    return func(inst, *args, **kwargs)
  File "/opt/odoo/odoo/tools/lru.py", line 34, in __getitem__
    a = self.d[obj]
KeyError: ('account.move', <function Model._get_view_cache at 0x7fca6f97b1c0>, (False, 'form', None, 'en_US'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/odoo/odoo/http.py", line 1770, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "/opt/odoo/odoo/service/model.py", line 133, in retrying
    result = func()
  File "/opt/odoo/odoo/http.py", line 1797, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "/opt/odoo/odoo/http.py", line 2001, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "/opt/odoo/odoo/addons/base/models/ir_http.py", line 222, in _dispatch
    result = endpoint(**request.params)
  File "/opt/odoo/odoo/http.py", line 725, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/opt/odoo/addons/web/controllers/dataset.py", line 24, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/opt/odoo/addons/web/controllers/dataset.py", line 20, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 464, in call_kw
    result = _call_kw_model(method, model, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 435, in _call_kw_model
    result = method(recs, *args, **kwargs)
  File "/opt/odoo/odoo/addons/base/models/ir_ui_view.py", line 2442, in get_views
    result['views'] = {
  File "/opt/odoo/odoo/addons/base/models/ir_ui_view.py", line 2443, in <dictcomp>
    v_type: self.get_view(
  File "/opt/odoo/odoo/addons/base/models/ir_ui_view.py", line 2628, in get_view
    result = dict(self._get_view_cache(view_id, view_type, **options))
  File "<decorator-gen-51>", line 2, in _get_view_cache
  File "/opt/odoo/odoo/tools/cache.py", line 104, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/opt/odoo/odoo/addons/base/models/ir_ui_view.py", line 2584, in _get_view_cache
    arch, view = self._get_view(view_id, view_type, **options)
  File "/mnt/data/odoo-addons-dir/account_commission/models/account_move.py", line 97, in _get_view
    invoice_xml = etree.XML(arch)
  File "src/lxml/etree.pyx", line 3231, in lxml.etree.XML
  File "src/lxml/parser.pxi", line 1912, in lxml.etree._parseMemoryDocument
ValueError: can only parse strings

The above server error caused the following client error:
OwlError: An error occured in the owl lifecycle (see this Error's "cause" property)
    at handleError (http://oca-commission-17-0-pr524-ed32b2bf7b40.runboat.odoo-community.org/web/assets/00afc54/web.assets_web.min.js:916:101)
    at App.handleError (http://oca-commission-17-0-pr524-ed32b2bf7b40.runboat.odoo-community.org/web/assets/00afc54/web.assets_web.min.js:1542:29)
    at ComponentNode.initiateRender (http://oca-commission-17-0-pr524-ed32b2bf7b40.runboat.odoo-community.org/web/assets/00afc54/web.assets_web.min.js:1006:19)

Caused by: RPC_ERROR: Odoo Server Error
    at makeErrorFromResponse (http://oca-commission-17-0-pr524-ed32b2bf7b40.runboat.odoo-community.org/web/assets/00afc54/web.assets_web.min.js:2870:163)
    at XMLHttpRequest.<anonymous> (http://oca-commission-17-0-pr524-ed32b2bf7b40.runboat.odoo-community.org/web/assets/00afc54/web.assets_web.min.js:2874:13)

**Comment by jimhuds** [comment]
Getting the following error:

```
RPC_ERROR
Odoo Server Error
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/tools/cache.py", line 99, in lookup
    r = d[key]
  File "<decorator-gen-5>", line 2, in __getitem__
  File "/usr/lib/python3/dist-packages/odoo/tools/func.py", line 87, in locked
    return func(inst, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/tools/lru.py", line 34, in __getitem__
    a = self.d[obj]
KeyError: ('account.move', <function Model._get_view_cache at 0xf5ce83a80b80>, (585, 'tree', None, 'en_GB', None))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 1768, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "/usr/lib/python3/dist-packages/odoo/service/model.py", line 133, in retrying
    result = func()
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 1795, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 1999, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_http.py", line 222, in _dispatch
    result = endpoint(**request.params)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 725, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/dataset.py", line 24, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/dataset.py", line 20, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 464, in call_kw
    result = _call_kw_model(method, model, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 435, in _call_kw_model
    result = method(recs, *args, **kwargs)
  File "/mnt/enterprise-addons/web_studio/models/ir_ui_view.py", line 53, in get_views
    return super().get_views(views, options)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 2441, in get_views
    result['views'] = {
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 2442, in <dictcomp>
    v_type: self.get_view(
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 2627, in get_view
    result = dict(self._get_view_cache(view_id, view_type, **options))
  File "<decorator-gen-51>", line 2, in _get_view_cache
  File "/usr/lib/python3/dist-packages/odoo/tools/cache.py", line 104, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 2583, in _get_view_cache
    arch, view = self._get_view(view_id, view_type, **options)
TypeError: AccountMove._get_view() got an unexpected keyword argument 'action_id'
```

**Comment by Tisho99** [comment]
Hi @jimhuds @Jaimermaccione 

I have fixed that error

Thank you for the review

**Comment by Tisho99** [comment]
Hi @pedrobaeza 

Do you remember why the "{'partner_id': partner_id}" context is needed in the invoice_line_ids field of the account.move form view? It is added with the fields_view_get function.

In v16 the context is not present because [the function](https://github.com/OCA/commission/blob/9e3748ffc3afc84e7b53bd47d281dd018b87fd1c/account_commission/models/account_move.py#L91) is badly inherited, and no one has complained about it. Does it also need to be fixed in v16?

**Review by Jaimermaccione — APPROVED** [review]
LGTM! Functional review on runboat

**Comment by pedrobaeza** [comment]
Searching for it, I don't find the reason, so it's reasonable to be removed and check if any side effect for not dragging dead code.

File: account_commission/views/report_settlement_templates.xml [hunk 1]
```diff
@@ -12,30 +12,18 @@
         </xpath>
         <xpath expr="//th" position="after">
             <t t-if="o.settlement_type == 'sale_invoice'">
-                <th class="text-start">Invoice</th>
-                <th class="text-start">Customer</th>
-                <th class="text-start">Description</th>
+                <th class="text-start">Invoice line</th>
                 <th class="text-start">Quantity</th>
-                <th class="text-start">Amount</th>
```
Comment (CarlosRoca13) [inline account_commission/views/report_settlement_templates.xml]: Why has the information shown in the report been changed?

Comment (pedrobaeza) [reply account_commission/views/report_settlement_templates.xml]: cc @Tisho99 @HaraldPanten

Comment (HaraldPanten) [reply account_commission/views/report_settlement_templates.xml]: @ValentinVinagre

Comment (Tisho99) [reply account_commission/views/report_settlement_templates.xml]: Hello @CarlosRoca13 

I do not remember now why that was changed. I forward ported another PR, and this changes were propagated from there, I may have missed this mistake

Seeing it now, it seems an error. Do you need to revert it?

**Pull request merged**